docs/en/resources/introduction.md

```diff
@@ -1,141 +1,83 @@
+| **Sharable Resource Options**             | Resource options can be shared between widgets via the resource middleware, allowing multiple widgets to react to resource changes.                          |
 
-# Basic usage
+# Basic Usage
 
-A Dojo resource can be created to fetch data in some way, and widgets are then able to access data from the resource through the data middleware.
+Dojo resources are created using the `createResourceTemplate` factory from `@dojo/framework/middleware/resources`. The default template comes with built-in support for working with in-memory data that can be used with data-aware widgets. This template can be used with "resource aware" widget, meaning any widget that uses the `resource` middleware from `@dojo/framework/core/middleware/resources`.
 
-## Creating a resource
-
-A basic resource can be created using the `createResource` factory from `@dojo/framework/core/resource` module, configured to work with in-memory data that can be passed within a widget.
-
-> resources.tsx
-
-```tsx
-const resource = createResource();
-```
-
-This `resource` can be used with any data aware widget, with any data that is passed.
-
-> MyWidget.tsx
+> App.tsx
 
 ```tsx
 import { create, tsx } from '@dojo/framework/core/vdom';
-
-import MyDataAwareWidget from './MyDataAwareWidget';
-import myResource from './resources';
+import myResource from '@dojo/framework/core/middleware/resources';
+import DataAwareWidget from './DataAwareWidget';
 
 const factory = create();
+const template = createResourceTemplate<{ foo: string }>();
 
-const MyWidget = factory(function MyWidget() {
-	return (
-		<MyDataAwareWidget
-			resource={myResource([
-				/* array of data for the widget to use */
-			])}
-		/>
-	);
+const App = factory(function App() {
+	return <DataAwareWidget resource={template({ data: [{ foo: 'string' }] })} />;
 });
 ```
 
-## Custom Data Template
+# Customizing a template's data source
 
-To use a `resource` to use a data source instead of requiring data to be passed into the resource on usage, a custom `DataTemplate` is required. By default the `DataTemplate` provided is an in-memory one that needs `data` to be passed on usage. The following is an example of a `DataTemplate` that wires a resource to a remote REST API.
+Dojo resources can be configured for a user defined data source, for example a RESTful API. This is done by passing an object with a `read` function that is responsible for fetching and setting the external data.
 
-> main.ts
+> userResourceTemplate.tsx
 
-```ts
-import { createResource } from '@dojo/framework/core/resource';
+```tsx
+import { createResourceTemplate } from '@dojo/framework/core/middleware/resources';
 
-const resource = createResource({
-	read: async (options: ReadOptions) => {
+export default createResourceTemplate({
+	read: async (request: ResourceReadRequest, controls: ResourceControls) => {
```

Comment:
I don't follow what `request` is here or why it is passed to the `controls.put` function at the end of this example

Reply:
I've missed some changes required in the comments of the example that I will update. the request is the details of the read request? like request/response naming. The request needs to be passed to the controls so that it knows what the response is for.